bescort.lic

```diff
@@ -448,6 +455,58 @@ class Bescort
     end
   end
 
+  def take_sandbarge(mode, mode_2)
+    case mode
+    when 'muspari'
+	  if mode_2 == 'oasis'
+	    manual_go2(6872)
+	    port_type = 'oasis'
+        port_call = 'The sand barge pulls up to a desert oasis'
+	  elsif mode_2 == 'hvaral'
+		manual_go2(6872)
+		port_type = 'dock'
+		port_call = 'The sand barge pulls into dock'
+	  else
+		echo('You entered the same locations.')
+        exit
+	  end
+    when 'oasis'
+	  if mode_2 == 'hvaral'
+	    manual_go2(7578)
+        port_type = 'dock'
+	    port_call = 'The sand barge pulls into dock'
+      elsif mode_2 == 'muspari'
+	    manual_go2(7579)
+        port_type = 'platform'
+	    port_call = 'The sand barge pulls into dock'
+	  else
+		echo('You entered the same locations.')
+        exit
+	  end
+	when 'hvaral'
+	  if mode_2 == 'muspari'
+	    manual_go2(3766)
+        port_type = 'platform'
+        port_call = 'The sand barge pulls into dock'
+      elsif mode_2 == 'oasis'
+	    manual_go2(3766)
+	    port_type = 'oasis'
+        port_call = 'The sand barge pulls up to a desert oasis'
+	  else
+		echo('You entered the same locations.')
+        exit
+	  end
+    end
+    case bput('go barge', 'What were you referring to', "One of the barge's crew members watching", "You can't do that right now")
+    when /What were you referring to/, /You can't do that right now/
+      waitfor 'A sand barge pulls'
+	  take_sandbarge(mode, mode_2)
+    when /One of the barge's crew members watching/
+      waitfor "#{port_call}"
+	  fput("go #{port_type}")
```

Comment:
You can use `bput` just like you did on line 486. It'll handle roundtime internally better than `fput` does.